crates/aptos/src/common/utils.rs

```diff
@@ -190,3 +200,68 @@ pub async fn get_sequence_number(
     let account = account_response.inner();
     Ok(account.sequence_number)
 }
+
+/// Error message for parsing a map
+const PARSE_MAP_SYNTAX_MSG: &str = "Invalid syntax for map.  Example: Name=Value,Name2=Value";
+
+/// Parses an inline map of values
+///
+/// Example: Name=Value,Name2=Value
+pub fn parse_map<K: FromStr + Ord, V: FromStr>(str: &str) -> anyhow::Result<BTreeMap<K, V>>
+where
+    K::Err: 'static + std::error::Error + Send + Sync,
+    V::Err: 'static + std::error::Error + Send + Sync,
+{
+    let mut map = BTreeMap::new();
+
+    // Split pairs by commas
+    for pair in str.split_terminator(',') {
+        // Split pairs by = then trim off any spacing
+        let (first, second): (&str, &str) = pair
+            .split_terminator('=')
+            .collect_tuple()
+            .ok_or_else(|| anyhow::Error::msg(PARSE_MAP_SYNTAX_MSG))?;
+        let first = first.trim();
+        let second = second.trim();
+        if first.is_empty() || second.is_empty() {
+            return Err(anyhow::Error::msg(PARSE_MAP_SYNTAX_MSG));
+        }
+
+        // At this point, we just give error messages appropriate to parsing
+        let key: K = K::from_str(first)?;
+        let value: V = V::from_str(second)?;
+        map.insert(key, value);
+    }
+    Ok(map)
+}
+
+/// Submits a [`TransactionPayload`] as signed by the `sender_key`
+pub async fn submit_transaction(
+    url: Url,
+    chain_id: ChainId,
+    sender_key: Ed25519PrivateKey,
+    payload: TransactionPayload,
+    max_gas: u64,
+) -> CliTypedResult<Transaction> {
+    let client = Client::new(url);
+
+    // Get sender address
+    let sender_address = AuthenticationKey::ed25519(&sender_key.public_key()).derived_address();
+    let sender_address = AccountAddress::new(*sender_address);
+
+    // Get account to get the sequence number
```

Comment:
nit: "Get account to get the sequence number"? Why not just "Get sequence number for address"?